compiler_opt/tools/combine_tfa_policies_lib_test.py

```diff
@@ -0,0 +1,151 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the combine_tfa_policies_lib.py module"""
+
+from absl.testing import absltest
+
+import tensorflow as tf
+from compiler_opt.tools import combine_tfa_policies_lib
+from tf_agents.trajectories import time_step as ts
+import tf_agents
+from tf_agents.specs import tensor_spec
+from tf_agents.trajectories import policy_step
+from tf_agents.typing import types
+import hashlib
+import numpy as np
+
+
+def client_side_model_selector_calculation(policy_name: str) -> types.Tensor:
+  m = hashlib.md5()
+  m.update(policy_name.encode('utf-8'))
+  high = int.from_bytes(m.digest()[8:], 'little')
+  low = int.from_bytes(m.digest()[:8], 'little')
+  model_selector = tf.constant([[high, low]], dtype=tf.uint64)
+  return model_selector
+
+
+class AddOnePolicy(tf_agents.policies.TFPolicy):
+  """Test policy which increments the obs feature."""
+
+  def __init__(self):
+    obs_spec = {'obs': tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)}
+    time_step_spec = ts.time_step_spec(obs_spec)
+
+    act_spec = tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)
+
+    super().__init__(time_step_spec=time_step_spec, action_spec=act_spec)
+
+  def _distribution(self, time_step):
+    """Boilerplate function for TFPolicy."""
+    pass
+
+  def _variables(self):
+    """Boilerplate function for TFPolicy."""
+    return ()
+
+  def _action(self, time_step, policy_state, seed):
+    """Boilerplate function for TFPolicy."""
+    observation = time_step.observation['obs'][0]
+    action = tf.reshape(observation + 1, (1,))
+    return policy_step.PolicyStep(action, policy_state)
+
+
+class SubtractOnePolicy(tf_agents.policies.TFPolicy):
+  """Test policy which decrements the obs feature."""
+
+  def __init__(self):
+    obs_spec = {'obs': tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)}
+    time_step_spec = ts.time_step_spec(obs_spec)
+
+    act_spec = tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)
+
+    super().__init__(time_step_spec=time_step_spec, action_spec=act_spec)
+
+  def _distribution(self, time_step):
+    """Boilerplate function for TFPolicy."""
+    pass
+
+  def _variables(self):
+    """Boilerplate function for TFPolicy."""
+    return ()
+
+  def _action(self, time_step, policy_state, seed):
+    """Boilerplate function for TFPolicy."""
+    observation = time_step.observation['obs'][0]
+    action = tf.reshape(observation - 1, (1,))
+    return policy_step.PolicyStep(action, policy_state)
+
+
+observation_spec = ts.time_step_spec({
+    'obs':
+        tf.TensorSpec(dtype=tf.int32, shape=(), name='obs'),
+    'model_selector':
+        tf.TensorSpec(shape=(2,), dtype=tf.uint64, name='model_selector')
+})
+
+action_spec = tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)
+
+
+class FeatureImportanceTest(absltest.TestCase):
```

Comment:
This class name doesn't seem particularly relevant and should probably be switched?